src/redux/reducers/todo.reducer.js

```diff
@@ -0,0 +1,17 @@
+import * as AT from '../action.types';
+
+const INITIAL_STATE = {dataStream: {label: 'Instagram'}, rangeDate: {startDate: {}, endDate: {}}};
+const todoReducer = (state = INITIAL_STATE, action) => {
+	switch (action.type) {
+		case AT.DATA_STREAM_REQUEST:
+			console.log(action.payload);
+			return {...state, dataStream: action.payload.dataStream};
+		
+		case AT.RANGE_DATE_REQUEST:
+			console.log(action.payload);
+			return {...state, rangeDate: action.payload.rangeDate};
```

Comment:
same here you can use set. this will work also but if you need to update a nested object it's easier using set.
for ex if you need to update - a.b.c.d.e.f in an object